### PR TITLE
Backport #12783: macOS: disable AppNap during sync (and mixing)

### PR DESCRIPTION
**Pull request by UdjinM6**
bitcoin#12783 1e0f3c44992fb82e6bf36c2ef9277b0759c17c4c macOS: disable AppNap during sync (Alexey Ivanov)

Original pull request description:
```
  Code based on pull/5804. Tested only on macOS 10.13.3 and should support 10.9+.

  What macOS versions bitcoin core currently supports?

Tree-SHA512: 85809b8d8d8a05169437b4268988da0b7372c29c6da3223ebdc106dc16dcb6d3caa5c52ace3591467005b50a63fd8b2ab1cb071cb4f450032932df25d5063315
```

The same issue is affecting mixing (which is quite annoying), so I modified it accordingly. Tested on 10.14.5 and it seems to be working as expected. Images below are for `this patch` (2192), `official 0.14.0.2 app` (2201) and `current develop` (2203), all started in regtest mode at ~ the same time.

After a minute or so, not synced:

Screenshot 2019-07-11 at 08 27 23

Forced sync (`mnsync next`), generated 1 block, waited ~1 minute:

Screenshot 2019-07-11 at 08 46 27

Started mixing, generated 1 more block, waited ~1 minute:

Screenshot 2019-07-11 at 08 52 42

### CODE REVIEW

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -950,6 +956,15 @@ void BitcoinGUI::updateHeadersSyncProgressLabel()
 
 void BitcoinGUI::setNumBlocks(int count, const QDateTime& blockDate, double nVerificationProgress, bool header)
 {
+#ifdef Q_OS_MAC
```
Comment (codablock) [inline src/qt/bitcoingui.cpp]: Oh my... :D
The original from Bitcoin was already bad...using the ternary operator as a replacement for if/else and at the same time relying on the side effects of the evaluation result... :D
Can you maybe change this to something as simple as:
```
#ifdef Q_OS_MAC
    bool enableAppNap = !masternodeSync.IsSynced();
#ifdef ENABLE_WALLET
    if (privateSendClient.fPrivateSendRunning) {
        enableAppNap = true;
    }
#endif
    if (enableAppNap) {
        m_app_nap_inhibitor->enableAppNap()
    } else {
        m_app_nap_inhibitor->disableAppNap()
    }
#edif
```

Comment (codablock) [reply src/qt/bitcoingui.cpp]: Updated the code sample to use disableAppNap instead of enableAppNap

**Comment by UdjinM6** [comment]
Force-pushed to trigger Travis build (failure seems to be unrelated https://travis-ci.org/dashpay/dash/jobs/557206151#L2645 but just in case).

File: src/qt/bitcoingui.h [hunk 2]
```diff
@@ -19,6 +19,12 @@
 #include <QPushButton>
 #include <QSystemTrayIcon>
 
+#ifdef Q_OS_MAC
+#include <qt/macos_appnap.h>
+#endif
+
+#include <memory>
```
Comment (PastaPastaPasta) [inline src/qt/bitcoingui.h]: is this addition needed? the bitcoin PR didn't add this

Comment (UdjinM6) [reply src/qt/bitcoingui.h]: Hmmm, true. Not sure how it got here 🙈 Dropped.
